clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/UpsertSecurityGroupAtomicOperation.groovy

```diff
@@ -112,4 +109,19 @@ class UpsertSecurityGroupAtomicOperation implements AtomicOperation<Void> {
     null
   }
 
+  private ConvertedIngress convertDescriptionToIngress(SecurityGroupLookup securityGroupLookup, UpsertSecurityGroupDescription description, boolean failOnMissingSelfReferencingRules) {
+    ConvertedIngress ipPermissionsFromDescription = SecurityGroupIngressConverter.
+      convertIngressToIpPermissions(securityGroupLookup, description)
+
+    if (ipPermissionsFromDescription.missingSecurityGroups.hasMissingNonSelfReferencingGroups() && !failOnMissingSelfReferencingRules) {
```

Comment:
Is this logic correct ... throw an exception if we have missing non-self referential groups and `failOnMissingSelfReferential == false`?

Reply:
Yeah, not sure where my head was at there. Updated.